[websocketpp] build failure

### Operating system

windows 11

### Compiler

_No response_

### Steps to reproduce the behavior

```Shell
vcpkg update
C:\src\vcpkg\vcpkg>vcpkg install websocketpp
```


### Failure logs

Computing installation plan...
The following packages will be built and installed:
  * openssl:x64-windows@3.2.0#2
    websocketpp[core,recommended]:x64-windows@0.8.2#3
Additional packages (*) will be modified to complete this operation.
Detecting compiler hash for triplet x64-windows...
Restored 0 package(s) from C:\Users\43833\AppData\Local\vcpkg\archives in 122 us. Use --debug to see more details.
Installing 1/2 openssl:x64-windows@3.2.0#2...
Building openssl:x64-windows@3.2.0#2...
-- Using cached openssl-openssl-openssl-3-aaf4f13b.2.0.tar.gz.
-- Cleaning sources at C:/src/vcpkg/vcpkg/buildtrees/openssl/src/nssl-3.2.0-5a1db6e780.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/src/vcpkg/vcpkg/downloads/openssl-openssl-openssl-3-aaf4f13b.2.0.tar.gz
-- Applying patch asm-comments.patch
-- Applying patch declspec-align.patch
-- Applying patch qt-msvc.patch
-- Applying patch script-prefix.patch
-- Applying patch windows/install-layout.patch
-- Applying patch windows/install-pdbs.patch
-- Applying patch unix/android-cc.patch
-- Applying patch unix/move-openssldir.patch
-- Applying patch unix/no-empty-dirs.patch
-- Applying patch unix/no-static-libs-for-shared.patch
-- Applying patch unix/disable-quicserver.patch
-- Using source at C:/src/vcpkg/vcpkg/buildtrees/openssl/src/nssl-3.2.0-5a1db6e780.clean
-- Found external ninja('1.11.0').
-- Getting CMake variables for x64-windows
-- Getting CMake variables for x64-windows
-- Prerunning x64-windows-dbg
-- Building x64-windows-dbg
-- Restarting build without parallelism
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: "V:/software/visual studio/VC/Tools/MSVC/14.38.33130/bin/Hostx64/x64/nmake.exe" /NOLOGO /G /U /F makefile install_dev install_modules INSTALL_PDBS=ON
    Working Directory: C:/src/vcpkg/vcpkg/buildtrees/openssl/x64-windows-dbg
    See logs for more information:
      C:\src\vcpkg\vcpkg\buildtrees\openssl\install-x64-windows-dbg-nmake-out.log
      C:\src\vcpkg\vcpkg\buildtrees\openssl\install-x64-windows-dbg-nmake-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_nmake.cmake:156 (vcpkg_execute_build_process)
  ports/openssl/windows/portfile.cmake:69 (vcpkg_build_nmake)
  ports/openssl/portfile.cmake:78 (include)
  scripts/ports.cmake:170 (include)


error: building openssl:x64-windows failed with: BUILD_FAILED
Elapsed time to handle openssl:x64-windows: 1 min
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+openssl
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?title=[openssl]+Build+error+on+x64-windows&body=Copy+issue+body+from+C%3A%2Fsrc%2Fvcpkg%2Fvcpkg%2Finstalled%2Fvcpkg%2Fissue_body.md


install-x64-windows-dbg-nmake-out.log
	"V:\software\visual studio\VC\Tools\MSVC\14.38.33130\bin\Hostx64\x64\nmake.exe" /LU                 depend
	"V:\software\visual studio\VC\Tools\MSVC\14.38.33130\bin\Hostx64\x64\nmake.exe" /LU                 _build_libs
	"V:\software\visual studio\VC\Tools\MSVC\14.38.33130\bin\Hostx64\x64\cl.exe"  /Zi /Fdossl_static.pdb /Gs0 /GF /Gy /MDd -nologo -DWIN32 -D_WINDOWS -W3 -utf-8 -MP  -D_DEBUG -MDd -Z7 -Ob0 -Od -RTC1 -I"crypto" -I"." -I"include" -I"providers\common\include" -I"providers\implementations\include" -D"L_ENDIAN" -D"OPENSSL_PIC" -D"OPENSSLDIR=\"C:\\src\\vcpkg\\vcpkg\\packages\\openssl_x64-windows\\debug\"" -D"ENGINESDIR=\"C:\\src\\vcpkg\\vcpkg\\packages\\openssl_x64-windows\\debug\\lib\\engines-3\"" -D"MODULESDIR=\"C:\\src\\vcpkg\\vcpkg\\packages\\openssl_x64-windows\\debug\\bin\"" -D"OPENSSL_BUILDING_OPENSSL" -D"OPENSSL_SYS_WIN32" -D"WIN32_LEAN_AND_MEAN" -D"UNICODE" -D"_UNICODE" -D"_CRT_SECURE_NO_DEPRECATE" -D"_WINSOCK_DEPRECATED_NO_WARNINGS" -D"DEBUG" -D"_DEBUG"   -DAES_ASM -DBSAES_ASM -DCMLL_ASM -DECP_NISTZ256_ASM -DGHASH_ASM -DKECCAK1600_ASM -DMD5_ASM -DOPENSSL_BN_ASM_GF2m -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_CPUID_OBJ -DOPENSSL_IA32_SSE2 -DOPENSSL_USE_APPLINK -DPADLOCK_ASM -DPOLY1305_ASM -DRC4_ASM -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DVPAES_ASM -DWHIRLPOOL_ASM -DX25519_ASM -c /Focrypto\libcrypto-shlib-cversion.obj "crypto\cversion.c"
cversion.c
crypto\cversion.c(60): warning C4129: “V”: 不可识别的字符转义序列
crypto\cversion.c(60): warning C4129: “T”: 不可识别的字符转义序列
crypto\cversion.c(60): warning C4129: “M”: 不可识别的字符转义序列
crypto\cversion.c(60): warning C4129: “H”: 不可识别的字符转义序列
crypto\cversion.c(60): warning C4129: “c”: 不可识别的字符转义序列
crypto\cversion.c(60): error C2143: 语法错误: 缺少“;”(在“常数”的前面)
crypto\cversion.c(60): error C2143: 语法错误: 缺少“;”(在“字符串”的前面)

err.log
cl: 鍛戒护琛?warning D9025 :姝ｅ湪閲嶅啓鈥?Zi鈥?鐢ㄢ€?Z7鈥?
NMAKE : fatal error U1077: “"V:\software\visual studio\VC\Tools\MSVC\14.38.33130\bin\Hostx64\x64\cl.exe"  /Zi /Fdossl_static.pdb /Gs0 /GF /Gy /MDd -nologo -DWIN32 -D_WINDOWS -W3 -utf-8 -MP  -D_DEBUG -MDd -Z7 -Ob0 -Od -RTC1 -I"crypto" -I"." -I"include" -I"providers\common\include" -I"providers\implementations\include" -D"L_ENDIAN" -D"OPENSSL_PIC" -D"OPENSSLDIR=\"C:\\src\\vcpkg\\vcpkg\\packages\\openssl_x64-windows\\debug\"" -D"ENGINESDIR=\"C:\\src\\vcpkg\\vcpkg\\packages\\openssl_x64-windows\\debug\\lib\\engines-3\"" -D"MODULESDIR=\"C:\\src\\vcpkg\\vcpkg\\packages\\openssl_x64-windows\\debug\\bin\"" -D"OPENSSL_BUILDING_OPENSSL" -D"OPENSSL_SYS_WIN32" -D"WIN32_LEAN_AND_MEAN" -D"UNICODE" -D"_UNICODE" -D"_CRT_SECURE_NO_DEPRECATE" -D"_WINSOCK_DEPRECATED_NO_WARNINGS" -D"DEBUG" -D"_DEBUG"   -DAES_ASM -DBSAES_ASM -DCMLL_ASM -DECP_NISTZ256_ASM -DGHASH_ASM -DKECCAK1600_ASM -DMD5_ASM -DOPENSSL_BN_ASM_GF2m -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_CPUID_OBJ -DOPENSSL_IA32_SSE2 -DOPENSSL_USE_APPLINK -DPADLOCK_ASM -DPOLY1305_ASM -DRC4_ASM -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DVPAES_ASM -DWHIRLPOOL_ASM -DX25519_ASM -c /Focrypto\libcrypto-shlib-cversion.obj "crypto\cversion.c"”: 返回代码“0x2”
Stop.
NMAKE : fatal error U1077: “"V:\software\visual studio\VC\Tools\MSVC\14.38.33130\bin\Hostx64\x64\nmake.exe" /LU                 _build_libs”: 返回代码“0x2”
Stop.






### Additional context

Hi,I tried to run cmd with administrator privileges but I still encountered this problem, it suggests that the file cversion.c is missing but I compiled it correctly and put it into global variables and still encountered the following problem, do you have a solution for this? I also re-downloaded the Openssl package from the official site to try and replace the vcpkg download, but I'm still having problems and it doesn't work.

--- a/ports/openssl/mkbuildinf.diff
+++ b/ports/openssl/mkbuildinf.diff
@@ -1,0 +1,15 @@
+diff --git a/util/mkbuildinf.pl b/util/mkbuildinf.pl
+index a57c80a..5cecf83 100755
+--- a/util/mkbuildinf.pl
++++ b/util/mkbuildinf.pl
+@@ -9,7 +9,9 @@
+ use strict;
+ use warnings;
+ 
+-my ($cflags, $platform) = @ARGV;
++my $platform = pop @ARGV;
++my $cflags = join(' ', @ARGV);
++$cflags =~ s(\\)(\\\\)g;
+ $cflags = "compiler: $cflags";
+ 
+ # Use the value of the envvar SOURCE_DATE_EPOCH, even if it's

--- a/ports/openssl/portfile.cmake
+++ b/ports/openssl/portfile.cmake
@@ -15,6 +15,7 @@ vcpkg_from_github(
     PATCHES
         cmake-config.patch
         command-line-length.patch
+        mkbuildinf.diff
         script-prefix.patch
         asm-armcap.patch
         windows/install-layout.patch

--- a/ports/openssl/vcpkg.json
+++ b/ports/openssl/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "openssl",
   "version": "3.4.0",
+  "port-version": 1,
   "description": "OpenSSL is an open source project that provides a robust, commercial-grade, and full-featured toolkit for the Transport Layer Security (TLS) and Secure Sockets Layer (SSL) protocols. It is also a general-purpose cryptography library.",
   "homepage": "https://www.openssl.org",
   "license": "Apache-2.0",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -6766,7 +6766,7 @@
     },
     "openssl": {
       "baseline": "3.4.0",
-      "port-version": 0
+      "port-version": 1
     },
     "opensubdiv": {
       "baseline": "3.5.0",

--- a/versions/o-/openssl.json
+++ b/versions/o-/openssl.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "facccb0b0c47f9804b6e336096d0985c3e541eec",
+      "version": "3.4.0",
+      "port-version": 1
+    },
+    {
       "git-tree": "ce504a83eb9627d54f1cffdb497a6bf5bd970d18",
       "version": "3.4.0",
       "port-version": 0

